### DISCUSSION TITLE
is the grammer based approach best approach to get structured data out of llama 7b ?

### DISCUSSION THREAD

**Discussion by vishugupta96** [ggml-org/llama.cpp#4102] [General]
Hi,
i am using llama 2 7B with grammar to turn unstructured data into structured json format
But i am unable find a lot of examples of this , i mean not a lot people are using this approach .
should i be using this for my project, 
as i cannot use chat gpt for security reasons is there a better way to  like:
using llama 13b with prompts without grammar or
some other entity extraction model
please suggest
thanks

**Comment by KerfuffleV2** [comment]
If you need to _guarantee_ that the result conforms to a grammar, I don't think you have a different alternative other than writing your own grammar sampler.

Otherwise you can try prompting the model to output in the format you want. If you're satisfied with the results, then you wouldn't need to use grammar sampling.
